networkx/algorithms/assortativity/pairs.py

```diff
@@ -1,26 +1,27 @@
-"""Generators of  x-y pairs of node data."""
+"""Generators of x-y pairs of node data."""
 import networkx as nx
 
 __all__ = ["node_attribute_xy", "node_degree_xy"]
 
 
 @nx._dispatchable(node_attrs="attribute")
 def node_attribute_xy(G, attribute, nodes=None):
-    """Returns iterator of node-attribute pairs for all edges in G.
+    """Yields a 2-tuple of (`attribute`, `attribute`) values for a specified
+    node-attribute for all edges in `G`.
```

Comment:
Can we make this the paragraph and have a shorter one-liner description for hyperlinks in the docs?
```suggestion
    """Yields 2-tuples of node attribute values for all edges in `G`.
    
    This generator yields, for each edge in `G` incident to a node in `nodes`,
    a 2-tuple of form ``(attribute value,  attribute value)`` for the parameter
    specified node-attribute.  for all edges in `G`.
```